scripts/process.py

```diff
@@ -1,61 +1,70 @@
+from dataflows import Flow, PackageWrapper, ResourceWrapper, validate
+from dataflows import add_metadata, dump_to_path, load, set_type, printer
 import os
 import urllib.request
 import logging
 import csv
 
 downloaded = 'cache/bbk_WU5500.csv'
-outpath = 'data/data.csv'
 source_url = 'http://www.bundesbank.de/cae/servlet/StatisticDownload?tsId=BBEX3.M.XAU.USD.EA.AC.C06&its_csvFormat=en&its_fileFormat=csv&mode=its'
 
-def download():
+
+def download_csv_resource():
+    source_url = 'http://www.bundesbank.de/cae/servlet/StatisticDownload?tsId=BBEX3.M.XAU.USD.EA.AC.C06&its_csvFormat=en&its_fileFormat=csv&mode=its'
+    downloaded_csv_filepath = 'cache/bbk_WU5500.csv'
     if not os.path.exists('cache'):
         os.makedirs('cache')
-    urllib.request.urlretrieve(source_url, downloaded)
+    urllib.request.urlretrieve(source_url, downloaded_csv_filepath)
+
+
+def rename(package: PackageWrapper):
+    package.pkg.descriptor['resources'][0]['name'] = 'gold-prices-monthly'
+    package.pkg.descriptor['resources'][0]['path'] = 'data/monthly.csv'
+    package.pkg.descriptor['resources'][1]['name'] = 'gold-prices-yearly'
+    package.pkg.descriptor['resources'][1]['path'] = 'data/yearly.csv'
+    yield package.pkg
+    res_iter = iter(package)
+    first: ResourceWrapper = next(res_iter)
+    second: ResourceWrapper = next(res_iter)
+    yield first.it
+    yield second.it
+    yield from package
+
 
-def extract():
+def fix_rows(rows):
+    # We skip top 5 rows and trim notes from the bottom. Also as dates are without day specified
+    # we are taking the first day in month by default
```

Comment:
And use `yearmonth` type for dates without dates

Reply:
FIXED